examples/golang/build.go

```diff
@@ -27,14 +27,16 @@ import (
 )
 
 //
-// configuration parameters can be set dynamically, like where to place the build directory
+// configuration parameters can be set dynamically, like where to place the logs/cache
 //
 
 var root = mustRoot(magnet.Config{
 	PrintConfig: true,
 	LogDir:      "_build/logs",
 	CacheDir:    "_build",
 	ModulePath:  "github.com/gravitational/magnet/examples/golang",
+	// If VERSION is unspecified, the default will be generated based on a git tag
+	Version: os.Getenv("VERSION"),
```

Comment:
This seems a bit reversed to me, the magnet.Config should be the version specified by the magnet.EnvVar and not the os.Getenv. So I think the normal way to do this is this line becomes `Version: version,`
and the line60 `version = root.E...` will automatically retrieve the os.Getenv to override any of the envvars magnet supports: https://github.com/gravitational/magnet/blob/master/env.go#L35

Reply:
Yes, this is a bit backwards but after I've removed the use of globals (which is this PR's primary goal), it would result in a circular dependency as `root` depends on `version` which depends on `root`. Maybe it would more "correct" to revert to the implicit globals in the magnet package for simplicity and instead remove the other bits that made use of the package cumbersome in some use-cases (e.g. implicit use of `make magnet-vars` to extract the otherwise custom configuration with no way of removing the reliance on the default `Makefile` file name, etc).